StyleTemplate_auto-generate/StyleTemplateCreator.bas

```diff
@@ -0,0 +1,806 @@
+    ''        End If
+    ''        If Not styleThis_style.ParagraphFormat.LineUnitAfter = False Then
+    ''            Debug.Print styleThis_style.NameLocal & " LineUnitAfter = " & styleThis_style.ParagraphFormat.LineUnitAfter
+    ''        End If
+    '
+    '    End If
+        '''' ------Checking char only styles?
+    '    If styleThis_style.Type = 2 Then
+    '       If styleThis_style.Linked = False Then
+    '               Debug.Print styleThis_style.NameLocal & " Linked = " & styleThis_style.Linked
+    '       End If
+    '    End if
+    
+    
+    '    ElseIf InStr(styleThis_style.NameLocal, "(") > 0 Then
+    '        Debug.Print "ALERT: Found a builtin style wiht a paren! :" & styleThis_style.NameLocal
+        End If
+        
+        If a Mod lngIncrement = 0 Or a = Word.ActiveDocument.Styles.Count Then
+            ActiveDocument.UndoClear
+        End If
+    Next
+    
+    strJson = JsonConverter.ConvertToJson(dictStyle_dict, Whitespace:=2)
+    OverwriteTextFile strJsonPath, strJson
+    Debug.Print "Scanned " & a & " styles, wrote info for " & dictStyle_dict.Count & " Macmillan Styles to json"
+
+End Sub
+
+Public Sub WriteNoColorTemplatefromJson()
+    ' So we can invoke WriteTemplatefromJson Sub for no color template, without rewriting hte whole thing over
+    Call WriteTemplatefromJsonCore(True)
+
+End Sub
+Public Sub WriteTemplatefromJson()
+    ' So we can invoke WriteTemplatefromJson Sub for regular template;
+    ' cuz optional parameters don't work in Word vba :(
+    Call WriteTemplatefromJsonCore(False)
+
+End Sub
+
+Private Function WriteTemplatefromJsonCore(Optional p_boolNoColor As Boolean = False)
+    ' NOTE: requirements for this sub are listed in comments under this moedule's "Declarations"
+    ' When invoked this sub looks for a "Styles.json" file in the same dir as
+    ' "StyleTemplateCreator.docm".  It loads all of the styles form that json file
+    ' and creates corresponding styles in a new macmillan.dotx template file
+    ' (also in the same dir).  Then it writes cycles through the new styles in the .dotx
+    ' and writes a paragraph for each one to the dotx.
+    ' It can be run from Mac for Word, but cannot write a handful of properties via Mac for Word:
+    ' The one that's really a problem is the Quickstyles property
+    Dim strJsonPath As String
+    Dim strNewFilePath As String
+    Dim strNewFileName As String
+    Dim dictStyle_dict As Dictionary
+    Dim a As Long
+    Dim b As Long
+    Dim c As Long
+    Dim n As Long
+    Dim lngIncrement As Long
+    Dim strStylename As String
+    Dim strStylename_B As String
+    Dim strStylename_C As String
+    Dim docTemplate As Document
+    Dim objBulletLT As ListTemplate
+    Dim objChecklistLT As ListTemplate
+    
+    ' file paths
+    strJsonPath = ThisDocument.Path & Application.PathSeparator & "Styles.json"
```

Comment:
Have the JSON file name match the main template file name ("macmillan.json"), we might have to revisit having multiple JSON files later.